great_expectations/experimental/datasources/dynamic_pandas.py

```diff
@@ -0,0 +1,284 @@
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Hashable,
+    Iterable,
+    List,
+    NamedTuple,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+)
+
+import pandas as pd
+import pydantic
+
+# https://github.com/pandas-dev/pandas/blob/main/pandas/_typing.py
+from pandas._typing import CompressionOptions, IndexLabel, StorageOptions
+from pydantic import FilePath
+
+# from pydantic.typing import resolve_annotations
+from typing_extensions import Final, Literal, TypeAlias
+
+from great_expectations.experimental.datasources.interfaces import DataAsset
+
+try:
+    from pandas._typing import CSVEngine
+except ImportError:
+    CSVEngine = Literal["c", "python", "pyarrow", "python-fwf"]
+
+logger = logging.getLogger(__file__)
+
+DataFrameFactoryFn: TypeAlias = Callable[..., pd.DataFrame]
+
+# sentinel values
+UNSUPPORTED_TYPE: Final = object()
+
+CAN_HANDLE: Final[Set[str]] = {
+    # builtins
+    "str",
+    "int",
+    "list",
+    "list[str]",
+    "set",
+    "tuple",
+    "dict",
+    "dict[str, str]",
+    "dict[str, list[str]]",
+    "dict[str, Any]",
+    "bool",
+    "None",
+    # typing
+    "Hashable",
+    "Sequence[Hashable]",
+    "Sequence[tuple[int, int]]",
+    "Sequence[Hashable]",
+    "Sequence[str]",
+    "Sequence[int]",
+    "Sequence[tuple[int, int]]",
+    "Literal['infer']",
+    "Literal[False]",
+    "Literal[True]",
+    "Literal['high', 'legacy']",
+    "Literal['frame', 'series']",
+    "Literal['xlrd', 'openpyxl', 'odf', 'pyxlsb']",
+    "Literal[None, 'header', 'footer', 'body', 'all']",
+    "Iterable[object]",
+    "Iterable[Hashable]",
+    # other
+    "Pattern",  # re
+    "Path",  # pathlib
+    "FilePath",  # pydantic
+    # pandas
+    "DtypeArg",
+    "CSVEngine",
+    "IndexLabel",
+    "CompressionOptions",
+    "StorageOptions",
+}
+
+NEED_SPECIAL_HANDLING: Dict[str, Set[str]] = defaultdict(set)
+FIELD_SKIPPED_UNSUPPORTED_TYPE: Set[str] = set()
+FIELD_SKIPPED_NO_ANNOTATION: Set[str] = set()
+
+
+class _SignatureTuple(NamedTuple):
+    name: str
+    signature: inspect.Signature
+
+
+class _FieldSpec(NamedTuple):
+    type: Type
+    default_value: object  # ... for required value
+
+
+FIELD_SUBSTITUTIONS: Final[Dict[str, Dict[str, _FieldSpec]]] = {
+    # CSVAsset
+    "filepath_or_buffer": {"path": _FieldSpec(pathlib.Path, ...)},
+    # JSONAsset
+    "path_or_buf": {"path": _FieldSpec(pathlib.Path, ...)},
+    # misc
+    "filepath": {"path": _FieldSpec(pathlib.Path, ...)},
+    "dtype": {"dtype": _FieldSpec(Optional[dict], None)},  # type: ignore[arg-type]
+    "dialect": {"dialect": _FieldSpec(Optional[str], None)},  # type: ignore[arg-type]
+    "usecols": {"usecols": _FieldSpec(Union[int, str, Sequence[int], None], None)},  # type: ignore[arg-type]
+    "skiprows": {"skiprows": _FieldSpec(Union[Sequence[int], int, None], None)},  # type: ignore[arg-type]
+}
+
+_METHOD_TO_CLASS_NAME_MAPPINGS: Final[Dict[str, str]] = {
+    "csv": "CSVAsset",
```

Comment:
@Kilo59 I would prefer that we spelled out `DataAsset` in the names -- and also formalized the capitalization as in the following examples:
`CsvDataAsset`

`JsonDataAsset`

`HtmlDataAsset`

etc.

There must be published guidelines for capitalization (I have seen some from the ".NET" literature). 

Thanks!